core/crypto/crypto.go

```diff
@@ -64,16 +64,13 @@ type Client interface {
 	GetEnrollmentCertificateHandler() (CertificateHandler, error)
 
 	// GetTCertHandlerNext returns a CertificateHandler whose certificate is the next available TCert
-	GetTCertificateHandlerNext() (CertificateHandler, error)
+	GetTCertificateHandlerNext(attributes []string) (CertificateHandler, error)
 
 	// GetTCertHandlerFromDER returns a CertificateHandler whose certificate is the one passed
-	GetTCertificateHandlerFromDER(der []byte) (CertificateHandler, error)
-
-	// ReadAttribute reads the attribute with name 'attributeName' from the der encoded x509.Certificate 'tcertder'.
-	ReadAttribute(attributeName string, tcertder []byte) ([]byte, error)
+	GetTCertificateHandlerFromDER(tCertDER []byte) (CertificateHandler, error)
 
 	// GetNextTCert gets next available (not yet used) transaction certificate.
-	GetNextTCert() (tCert, error)
+	GetNextTCert(attributes []string) (tCert, error)
```

Comment:
Changing the return value of this function is not in the scope of this PR (un?)fortunately. This PR  simply/mainly adds the `attributes` parameter to the function, not changing the returned value/type. Given that the existing interface returns a `tCert` already, it might be quicker/more efficient to review/focus on the current addition/diff and discuss other proposed changes to the API separately.